`vimPlugins.command-t` fails to build on darwin

The `command-t` vim plugin is currently undergoing a full rewrite. As part of this, it appears to have changed how it compiles. On Darwin it now compiles using `xcrun clang` ([Makefile#L37](https://github.com/wincent/command-t/blob/f6a6cf7fb17cee08ca0ca84ffce6aabccc3d8476/lua/wincent/commandt/lib/Makefile#L37)). `xcrun` is not available when building in Nix and so it fails. Most likely the reference to `xcrun` can just be patched out.

It also runs `getconf` ([Makefile#L8](https://github.com/wincent/command-t/blob/f6a6cf7fb17cee08ca0ca84ffce6aabccc3d8476/lua/wincent/commandt/lib/Makefile#L8)) which is also reporting a "command not found" line, although this appears to be non-fatal. This affects all platforms.

### Steps To Reproduce
Steps to reproduce the behavior:
1. build `vimPlugins.command-t`

### Build log
```
Sourcing vim-command-check-hook.sh
Using vimCommandCheckHook
Sourcing vim-gen-doc-hook
Sourcing neovim-require-check-hook.sh
Using neovimRequireCheckHook
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/gmicl6db5wb0hvaf0aprhbqkg81asca7-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
make -C lua/wincent/commandt/lib
make[1]: Entering directory '/private/tmp/nix-build-vimplugin-command-t-2022-09-18.drv-0/source/lua/wincent/commandt/lib'
bash: line 1: getconf: command not found
xcrun clang -Wall -Wextra -Wno-unused-parameter -DMAX_FILES_CONF=33554432 -DMMAP_SLAB_SIZE_CONF=34359738368 -DNDEBUG -O3 -DMACOS -shared -fPIC -o commandt.so commandt.c debug.c die.c find.c heap.c matcher.c scanner.c score.c str.c watchman.c xmalloc.c xmap.c xstrdup.c
bash: line 1: xcrun: command not found
make[1]: *** [Makefile:62: commandt.so] Error 127
make[1]: Leaving directory '/private/tmp/nix-build-vimplugin-command-t-2022-09-18.drv-0/source/lua/wincent/commandt/lib'
make: *** [Makefile:9: build] Error 2
```

### Notify maintainers
I don't know if anyone in particular is responsible here but @virchau13 appears to have fixed the build of this plugin most recently

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix run nixpkgs#nix-info -- -m
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.6`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.10.3`
 - channels(lily): `""`
 - channels(root): `""`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

--- a/pkgs/applications/editors/vim/plugins/overrides.nix
+++ b/pkgs/applications/editors/vim/plugins/overrides.nix
@@ -53,6 +53,7 @@
 , zsh
 
   # command-t dependencies
+, getconf
 , ruby
 
   # cpsm dependencies
@@ -239,10 +240,11 @@ self: super: {
   };
 
   command-t = super.command-t.overrideAttrs (old: {
-    buildInputs = [ ruby ];
+    nativeBuildInputs = [ getconf ruby ];
     buildPhase = ''
       substituteInPlace lua/wincent/commandt/lib/Makefile \
-        --replace '/bin/bash' 'bash'
+        --replace '/bin/bash' 'bash' \
+        --replace xcrun ""
       make build
       rm ruby/command-t/ext/command-t/*.o
     '';

